src/codegen/templates/_library_interpreter.py.mako

```diff
@@ -197,6 +211,9 @@ class LibraryInterpreter(BaseInterpreter):
         waveform_attribute_mode: WaveformAttributeMode
     ) -> int:
         """Read an analog waveform with timing and attributes."""
+        if number_of_samples_per_channel == READ_ALL_AVAILABLE:
+            number_of_samples_per_channel = self.get_default_number_of_samples_to_read(task_handle)
```

Comment:
This does not belong in the interpreter. All the other read methods do it in the reader:

https://github.com/ni/nidaqmx-python/blob/master/src/handwritten/stream_readers/_analog_multi_channel_reader.py#L88

https://github.com/ni/nidaqmx-python/blob/master/src/handwritten/task/_task.py#L282